### PR TITLE
CNF-13857: Add documention for configuring GitOps Flow with the SiteConfig Operator

### PR DESCRIPTION
**Pull request by sakhoury**

### CODE REVIEW

File: docs/argocd.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+resources:
+  - ns.yaml
+  - secrets.yaml
+  - extra-manifests-cm1.yaml
+  - extra-manifests-cm2.yaml
+  - clusterinstance.yaml
+```
+
+### Generating extra-manifests ConfigMap using kustomize
```
Comment (imiller0) [inline docs/argocd.md:61]: This section is an "alternative" or "advanced" option. Having it inline with the initial description of the flow makes it harder for the reader to follow the basic steps before they get into the "other options". I'd recommend having a section at the end for "advanced options" which includes this content.

File: docs/argocd.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+## Prepare Git Repository
+
+To start, create a Git repository with a directory structure similar to the suggested example shown below.
+
+```sh
+clusters/
+ |- spoke1/
+ |-- extra-manifests-cm1.yaml
+ |-- extra-manifests-cm2.yaml
```
Comment (imiller0) [inline docs/argocd.md:23]: Typically the extra manifests CM will be shared across many spoke clusters. Can you adjust these to reflect that?

File: docs/argocd.md (hunk 1 above)
Comment (sakhoury) [reply docs/argocd.md:61]: Moved to the end under a new Section `Advanced Topics`

File: docs/argocd.md (hunk 2 above)
Comment (sakhoury) [reply docs/argocd.md:23]: The extra manifests ConfigMaps need to be created in the same namespace as the `ClusterInstance` CR (i.e. local reference).

Comment (imiller0) [reply docs/argocd.md:25]: Doc is ok for now but this is a limitation that we will need to address in the dev/tech preview phases. The set of extra manifests will be consistent across many clusters (1000's) and we will need the ability to keep a central referenced version.
Can you please open a story to track this?

**Review by imiller0 — COMMENTED** [review]
lgtm

File: docs/argocd.md (hunk 2 above)
Comment (sakhoury) [reply docs/argocd.md:25]: @imiller0 I had created a [PR](https://github.com/stolostron/siteconfig/pull/7) to address this earlier on.

**Review by carbonin — APPROVED** [review]

**Pull request merged**